packages/core/src/lib/input/input.svelte

```diff
@@ -0,0 +1,57 @@
+<!--
+@component
+  
+For user inputs.
+
+This is the base input component that accepts all input properties without any 
+additional behaviors attached. Generally, other typed inputs like the 
+NumberInput or DateInput are preferable.
+
+```svelte
+<Input type="email"  on:click={onClick} />
+```
+-->
+<svelte:options immutable />
+
+<script lang="ts">
+import cx from 'classnames';
+
+/** Whether or not the input should be rendered as readonly. */
+export let readonly = false;
+
+/**
+ * Whether or not the input should be rendered as readonly and have
+ * `aria-disabled` set to true.
+ */
+export let disabled = false;
```

Comment:
Is there any reason to have separate props? It looks like these do the same things.

Reply:
Yes, mainly for accessibility. For readonly inputs, we generally still allow things like selecting and copying. We don't want to indicate to screen readers those inputs are disabled or inactionable. For disabled inputs, we generally do not allow any actions on the inputs, but they still need to be focusable so screen readers can see them. If we make the input disabled, it won't be focusable or perceivable to assistive devices. So we make it readonly so it can still be perceived, then set `aria-disabled` to indicate to them there is not any action they can take on the input other than being aware of it.

We could go a step further for disabled and assign a specific cursor as well, I went back and forth on that but may be useful for non-screenreader users to differentiate.

Reply:
@ethanlook OK so now `disabled` also sets `pointer-events-none` making it non-operable. I updated the prop docs to clarify: 

readonly = renders as readonly but is still operable
disabled = renders as readonly but is not operable